app/User.php

```diff
@@ -244,6 +244,49 @@ public function getGreetingAndName()
         return $this->getGreeting() . ' ' . $this->getFullName();
     }
 
+    /**
+     * Get the user's record by Billet ID. For use in the retrieval of
+     * Space Lords and heads of branches, where a single user holds the
+     * specified billet.
+     *
+     * @param string $billet_id
+     *
+     * @return User|null
+     */
+    public static function getByBilletId(string $billet_id)
+    {
+        $billet_user = null;
+
+        $billet = Billet::where('_id', $billet_id)->first();
+
+        if (isset($billet) && isset($billet['billet_name'])) {
+            $billet_user = User::where('assignment.billet', '=', $billet['billet_name'])
+                ->first();
+        }
+
```

Comment:
@cinnion repeat after me -- Defensive Programming :-)

We should check that the have a valid user.   Not just that we don't have a null, but that $billet_user's type is `App\Class`

Reply:
The only two things it will return is null or a user instance.